dev_guide/builds/advanced_build_operations.adoc

```diff
@@ -122,3 +122,147 @@ endif::[]
+         +-----+-----+
+               |
+               v
+       +---------------+       +--------------+
+       |   Image with  |       |    Runtime   |        +--------------+
+       |     Binary    |   +   |     Image    |   +    |  Dockerfile  |
+       |    Artifact   |       +-------+------+        +------+-------+
+       +-------+-------+               |                      |
+               |                       |                      |
+               |                       v                      |
+               |               +--------------+               |
+               |               | Docker Build |               |
+               +-------------->|     with     |<--------------+
+                               | Image Source |
+                               +--------------+
+                                       |
+                                       |
+                                       v
+                               +---------------+
+                               | Runtime Image |
+                               | with Compiled |
+                               |    Artifact   |
+                               +---------------+
+----
+
+The first build takes the application source and produces an image containing a
+WAR file. The image is pushed to the `artifact-image` image stream. The path of
+the output artifact will depend on the *_assemble_* script of the
+Source-to-Image builder used. In this case, it will be output to
+*_/wildfly/standalone/deployments/ROOT.war_*.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: BuildConfig
+metadata:
+  name: artifact-build
+spec:
+  output:
+    to:
+      kind: ImageStreamTag
+      name: artifact-image
+  source:
+    git:
+      uri: https://github.com/openshift/openshift-jee-sample.git
+    type: Git
+  strategy:
+    sourceStrategy:
+      from:
+        kind: ImageStreamTag
+        name: wildfly:10.1
+        namespace: openshift
+    type: Source
+----
+
+The second build uses xref:build_inputs.adoc#image-source[Image Source] with a
+path to the WAR file inside the output image from the first build. An inline
+*_Dockerfile_* copies that WAR file into a runtime image.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: BuildConfig
+metadata:
+  name: image-build
+spec:
+  output:
+    to:
+      kind: ImageStreamTag
+      name: image-build:latest
+  source:
+    type: Dockerfile
+    dockerfile: |-
+      FROM jee-runtime:latest
+      COPY ROOT.war /deployments/ROOT.war
+    images:
+    - from: <1>
+        kind: ImageStreamTag
+        name: artifact-image:latest
+        paths: <2>
+        - sourcePath: /wildfly/standalone/deployments/ROOT.war
+          destinationDir: "."
+  strategy:
+    dockerStrategy:
+      from: <3>
+        kind: ImageStreamTag
+        name: jee-runtime:latest
+    type: Docker
+  triggers:
+  - imageChange: {}
+    type: ImageChange
+----
+<1> `from` specifies that the Docker build should include the output of the image
```

Comment:
@csrwng In https://github.com/openshift/openshift-docs/pull/3785, this was "Dockerbuild" (one word). Just want to make sure this was meant to say "Docker build" and not "Dockerfile".